added initial tests and basic packaging

fernflower opened the pull request:
- Tests can be run by `pytest tests/`.
- Added setup.py so that ntp2chrony can be installed
  as a package via pip.

mlichvar commented:
Thanks for your work on this.

Can you please split the patch, one adding the packaging stuff, second adding a test framework and other adding specific tests?

As for testing itself, I'd like to have a simple way to specify arbitrary input files and compare it with expected output (possibly excluding comments). A simple shell script would be fine. I'm not sure if it's worth the trouble to test individual functions. What matters is the result.

We need to test multiple ntp configurations. A single configuration cannot test everything as there are exclusive paths in the generation of the chrony configs. If you include an example ntp.conf, please use something based on the RHEL/Fedora ntp.conf.

fernflower committed:
basic packaging

This will allow ntp2chrony be installed by pip

fernflower committed:
added initial unit tests

This commit contains basic unit tests covering conversion
details.
Functional tests in the form of various
ntp.conf -> chrony.conf with verification will be
added in the commit to follow.

fernflower committed:
Add initial ntp->chrony converter tests

Ntp configs to convert are stored in tests/data/ntpconfs dir,
all test case files have the same sequential number.
Expected chrony configs are stored in tests/data/chronyconfs.
For each test case ntp config is taken, conversion is run and the
result is compared with the expected result.

mlichvar commented:
Applied, thanks!
